File: src/CadastroIgreja.Domain/Entities/Igreja.cs
namespace CadastroIgreja.Domain.Entities;

public class Igreja
{
    public Guid Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;
    public Guid? ParentId { get; set; }
}

File: frontend/lib/src/models/igreja.dart
class Igreja {
  const Igreja({required this.id, required this.nome, required this.tipo, required this.ativa, this.parentId});

  final String id;
  final String nome;
  final String tipo;
  final String? parentId;
  final bool ativa;

  factory Igreja.fromJson(Map<String, Object?> json) {
    return Igreja(
      id: json['id'] as String? ?? '',
      nome: json['nome'] as String? ?? '',
      tipo: json['tipo'] as String? ?? '',
      parentId: json['parentId'] as String?,
      ativa: json['ativa'] as bool? ?? false,
    );
  }
}
